src/components/ErrorNotification/ErrorNotification.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect } from 'react';
+import classNames from 'classnames';
+import { ErrorType } from '../../types/ErrorType';
+
+type Props = {
+  errorType: ErrorType;
+  hasError: boolean;
+  onNotificationClose: () => void;
+};
+
+export const ErrorNotification: React.FC<Props> = React.memo(
+  ({
+    errorType,
+    hasError,
+    onNotificationClose,
+  }) => {
+    let errorMessage = '';
+
+    switch (errorType) {
+      case ErrorType.Add:
+      case ErrorType.Delete:
+      case ErrorType.Update:
+      case ErrorType.Load:
+        errorMessage = `Unable to ${errorType} a todo`;
+
+        break;
+
+      case ErrorType.EmptyTitle:
+        errorMessage = 'Title can\'t be empty';
+
+        break;
+
+      default:
+        errorMessage = '';
+    }
+
+    useEffect(() => {
+      const timerId = setTimeout(() => onNotificationClose(), 3000);
+
+      return () => clearTimeout(timerId);
+    });
```

Comment:
Giving useEffect no second argument acts as both componentDidMount and componentDidUpdate, as in it runs first on mount and then on every re-render.